### PR TITLE
tuba: init at 0.1.0

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Description of changes

[Tuba](https://tuba.geopjr.dev/) is a new GTK app for various Fediverse servers. It is the fork of now archived Mastodon application [Tootle](https://github.com/bleakgrey/tootle). I've been using this in my personal overlay for quite a while, and since there is a release now, I guess I'm going to maintain it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by infinitivewitch** [comment]
Result of `nixpkgs-review pr 222721` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tuba</li>
  </ul>
</details>

**Review by ghost — APPROVED** [review]

**Comment by Alizter** [comment]
Result of `nixpkgs-review pr 222721` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tuba</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]

File: pkgs/applications/misc/tuba/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    libgee
+    libsoup
+    gtk4
+    libadwaita
+    libsecret
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-libav
+    gst-plugins-base
+    (gst-plugins-good.override { gtkSupport = true; })
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tuba/default.nix:58]: Isn't this the default?

Comment (NickCao) [reply pkgs/applications/misc/tuba/default.nix:58]: Nope: https://github.com/NixOS/nixpkgs/blob/ca4a734f86c0c64d11d8910c910c31433219c878/pkgs/development/libraries/gstreamer/good/default.nix#L32

File: pkgs/applications/misc/tuba/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    hash = "sha256-dkURVzbDBrE4bBUvf2fPqvgLKE07tn7jl3OudZpEWUo=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    vala
+    python3
+    wrapGAppsHook
```
Comment (Aleksanaa) [inline pkgs/applications/misc/tuba/default.nix:40]: Use wrapGAppsHook4 instead?

Comment (chuangzhu) [reply pkgs/applications/misc/tuba/default.nix:40]: done in #224807

**Pull request merged**